shakenfist/daemons/net.py

```diff
@@ -148,16 +148,31 @@ def _process_network_node_workitems(self):
                 raise exceptions.UnknownTaskException(
                     'Network workitem was not decoded: %s' % workitem)
 
+            log_ctx = log_ctx.withNetwork(workitem.network_uuid())
             n = net.Network.from_db(workitem.network_uuid())
             if not n:
-                log_ctx.withNetwork(workitem.network_uuid()).warning(
-                    'Received work item for non-existent network')
+                log_ctx.warning('Received work item for non-existent network')
                 return
 
             # NOTE(mikal): there's really nothing stopping us from processing a bunch
             # of these jobs in parallel with a pool of workers, but I am not sure its
             # worth the complexity right now. Are we really going to be changing
             # networks that much?
+
+            # Tasks valid for a network in any state
+            if isinstance(workitem, RemoveDHCPNetworkTask):
+                n.remove_dhcp()
+                db.add_event('network', workitem.network_uuid(),
+                             'network node', 'remove dhcp', None, None)
+                return
+
+            # Tasks that should not operate on a dead network
+            if n.is_dead():
```

Comment:
Shouldn't we use the deleted and error states here? What is the added value of using is_dead() here?

Reply:
`is_dead()` includes the `deleting` state.

 That was the solution from a month or so ago to capture the in-flight delete case.

Reply:
Oh wait. I see now. I feel like is_ok(), is_created(), and is_dead() should probably move down to baseobject.py. There are lots of other places in the code which could use those calls (I'm mostly thinking instances here).

Reply:
I agree. I would like to see all the state comparisons to strings moved to `is_*` functions.